webdriver/newsession/invalid_capabilities.py

```diff
@@ -0,0 +1,88 @@
+#META: timeout=long
+
+import pytest
+from webdriver import error
+
+from conftest import product, flatten
+
+
+@pytest.mark.parametrize("value", [None, 1, "{}", []])
+def test_invalid_capabilites(new_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session({"capabilities": value})
+
+
+@pytest.mark.parametrize("value", [None, 1, "{}", []])
+def test_invalid_always_match(new_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session({"capabilities": {"alwaysMatch": value}})
+
+
+@pytest.mark.parametrize("value", [None, 1, "[]", {}])
+def test_invalid_first_match(new_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session({"capabilities": {"firstMatch": value}})
```

Comment:
Any empty collection is a valid value for `firstMatch` according to my reading of the spec. I suspect we're missing language to say that this should be expanded to a list with a single empty object.